cmd/skaffold/app/cmd/runner.go

```diff
@@ -71,11 +75,58 @@ func createNewRunner(opts config.SkaffoldOptions) (runner.Runner, []*latest.Skaf
 	return runner, configs, nil
 }
 
-func runContext(opts config.SkaffoldOptions) (*runcontext.RunContext, []*latest.SkaffoldConfig, error) {
+func runContext(out io.Writer, opts config.SkaffoldOptions) (*runcontext.RunContext, []*latest.SkaffoldConfig, error) {
+	configs, pipelines, err := skaffoldConfig(out, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: Should support per-config kubecontext. Right now we constrain all configs to define the same kubecontext.
+	kubectx.ConfigureKubeConfig(opts.KubeConfig, opts.KubeContext, configs[0].Deploy.KubeContext)
+
+	if err := validation.Process(configs); err != nil {
+		return nil, nil, fmt.Errorf("invalid skaffold config: %w", err)
+	}
+
+	runCtx, err := runcontext.GetRunContext(opts, pipelines)
+	if err != nil {
+		return nil, nil, fmt.Errorf("getting run context: %w", err)
+	}
+
+	if err := validation.ProcessWithRunContext(runCtx); err != nil {
+		return nil, nil, fmt.Errorf("invalid skaffold config: %w", err)
+	}
+
+	return runCtx, configs, nil
+}
+
+func setDefaultDeployer(configs []util.VersionedConfig) bool {
+	// set the default deployer only if no deployer is explicitly specified in any config
+	for _, cfg := range configs {
+		if cfg.(*latest.SkaffoldConfig).Deploy.DeployType != (latest.DeployType{}) {
+			return false
+		}
+	}
+	return true
+}
+
+// skaffoldConfig will try to parse the given opts.ConfigurationFile. If not found, it will try to automatically generate a config for the user
+func skaffoldConfig(out io.Writer, opts config.SkaffoldOptions) ([]*latest.SkaffoldConfig, []latest.Pipeline, error) {
 	parsed, err := schema.ParseConfigAndUpgrade(opts.ConfigurationFile, latest.Version)
 	if err != nil {
 		if os.IsNotExist(errors.Unwrap(err)) {
-			return nil, nil, fmt.Errorf("skaffold config file %s not found - check your current working directory, or try running `skaffold init`", opts.ConfigurationFile)
+			color.Default.Fprintf(out, "Skaffold config file %s not found - Trying to create one for you...\n", opts.ConfigurationFile)
+			config, err := initializer.Transparent(context.Background(), out, initConfig.Config{Opts: opts})
+			if err != nil {
+				return nil, nil, fmt.Errorf("unable to generate skaffold config file automatically - try running `skaffold init`: %w", err)
+			}
+			if config == nil {
+				return nil, nil, fmt.Errorf("user not continuing")
```

Comment:
should this be a bit more verbose?

Reply:
Is there anything specific you feel should be mentioned here? Maybe something like
> user not continuing with generated config

Reply:
I was thinking something more fatal like:
```go
return nil, nil, errors.New("unable to generate skaffold config file automatically - try running `skaffold init`: action cancelled by user")
```

Reply:
makes sense, I can change that